### ISSUE TITLE
Can I get this code with c++ version?

### ISSUE THREAD

**Issue by HKHK0929** [NVIDIA/VideoProcessingFramework#320]
Hello, I read your Nvidia blog which is about VPF.
Actually, I'm a student. And I make some program for graduate project.
I have some problem about latency because of video decoding.
I want to convert mp4 video into picture frame which is RGB file.
so I use nvdec. but when nvdec decode the demuxed video(only frames not audio), it turn nv12 format and go to cpu mem automatically.(GPU mem-> CPU mem). 
I have to use RGB format, so I move the nv12 format file to device memory and change into RGB format.
As you know, it takes a lot of time.
it is very inefficient to move the files Device to host. so I searched in google and found your blog.
my system is c++ environment. so I can't use python codes. 
So, can I get your code which is written with c++?
Thanks for reading my issue.

**Comment by rarzumanyan** [comment]
Hi @HKHK0929 

>So, can I get your code which is written with c++?

Yes you can. 
VPF core is written in C++ so you don't need to do anything specific.
Every time you build it C++ libraries are built + pybind11 glue which exports it to Python module.
If you want to use C++ API, you need to follow usual build procedure and link your application against `TC_CORE` and `TC` libraries.

You can reference source files from `PyNvCodec/src` folder to see how to use C++ API.
E. g. this is how it's used in `Surface` resizer class:
https://github.com/NVIDIA/VideoProcessingFramework/blob/335d8ba630feae00463a462ecc207c946a89eb79/PyNvCodec/src/PySurfaceResizer.cpp#L28-L53

Basically whole `PyNvCodec` can be used to illustrate how to use `TC` library API.

**Comment by HKHK0929** [comment]
Thanks a lot. I will try it.
I appreciate your help.
